### PR TITLE
refactor(fields): readability and performance improvements [CLK-510098]

### PR DESCRIPTION
**Pull request by mysza**
## Summary

- updated types names to better convey meaning
- improved dependencies featching
- removed unnecessary conditional code
- added ability to add variables in bulk (what we usually do in client code)

## Tests

- fixed custom fields generator to produce proper var names
- added "performance" test for graph generation
- separated tests for validation and dependencies tracking from graph generation overhead

### CODE REVIEW

File: src/clickup/clickupFieldsDependencyTracker.ts [hunk 1]
```diff
@@ -83,14 +91,14 @@ export class ClickUpFieldsDependencyTracker {
...
+            recStack: new Set<VariableName>(),
             depthByNode: {},
         };
 
-        const sameFieldInPath = (context: ValidationContext, fieldName: FieldName): boolean =>
+        const sameFieldInPath = (context: ValidationContext, fieldName: VariableName): boolean =>
             context.recStack.has(fieldName);
 
-        function traverseNode(fieldName: FieldName, context: ValidationContext) {
+        function traverseNode(fieldName: VariableName, context: ValidationContext) {
```
Comment (szygendab) [inline src/clickup/clickupFieldsDependencyTracker.ts:101]: nit: 
```suggestion
        function traverseNode(variableName: VariableName, context: ValidationContext) {
```

File: test/scripts/generatePerfTestData.ts [hunk 2]
```diff
@@ -33,41 +33,42 @@ function generateCustomFields(
     formulaRatio: number,
     dependenciesPerFormula: number,
     formulaDependencyRatio: number
-): CustomFieldVariable[] {
-    const fieldIds: FieldName[] = Array.from({ length: numFields }, (_, i) => `FIELD_${i}`);
-    const customFields = new CustomFieldsSet();
+): ClickUpParserVariable[] {
+    const makeVariableName = (id: number) => `VAR_${id}`;
```
Comment (szygendab) [inline test/scripts/generatePerfTestData.ts:37]: is that a new convention? Could we document it?

**Review by szygendab — APPROVED** [review]

File: test/scripts/generatePerfTestData.ts (hunk 2 above)
Comment (mysza) [reply test/scripts/generatePerfTestData.ts:37]: No it's not, in this case this is test data generation for dependency tracker "performance" tests, and dependency tracker doesn't care about the variable name, so anything works.
I'll update to `CUSTOM_FIELD_*` with a comment to make it clear 👍🏻

Comment (nikitaclicks) [reply test/scripts/generatePerfTestData.ts:37]: I think it is just for tests

**Review by nikitaclicks — APPROVED** [review]

**Pull request merged**